module/modmanager/manager.go

```diff
@@ -1075,6 +1076,51 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 	}
 }
 
+// FirstRun is runs a module-specific setup script.
+// TODO: return exit code?
+func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+	logger := mgr.logger.WithFields("name", conf.Name)
+
+	// Evaluate the Module's FirstRun path. If there is an error we assume
+	// that the first run script does not exist and we debug log and exit quietly.
+	firstRunPath, err := conf.EvaluateFirstRunPath(packages.LocalPackagesDir(mgr.packagesDir))
+	if err != nil {
+		logger.Debug("no first run script detected, skipping setup phase", "error", err)
+		return nil
```

Comment:
discussed IRL: we may want to distinguish between different kinds of errors in evaluating the first run path to determine whether to debug log or warn.

Reply:
if first run already succeeded, would be good to separately info log that a first_run script was detected but will not be run